contracts/JumpRateModelV2.sol

```diff
@@ -48,4 +117,105 @@ contract JumpRateModelV2 is BaseJumpRateModelV2 {
     ) external view override returns (uint256) {
         return _getBorrowRate(cash, borrows, reserves, badDebt);
     }
+
+    /**
+     * @notice Calculates the current supply rate per slot (block or second)
+     * @param cash The amount of cash in the market
+     * @param borrows The amount of borrows in the market
+     * @param reserves The amount of reserves in the market
+     * @param reserveFactorMantissa The current reserve factor for the market
+     * @param badDebt The amount of badDebt in the market
+     * @return The supply rate percentage per slot (block or second) as a mantissa (scaled by EXP_SCALE)
+     */
+    function getSupplyRate(
+        uint256 cash,
+        uint256 borrows,
+        uint256 reserves,
+        uint256 reserveFactorMantissa,
+        uint256 badDebt
+    ) public view virtual override returns (uint256) {
```

Comment:
We have two different use cases on the frontend.
1) We fetch the supply and borrow rates of each vToken by using the `allPools` method of the `PoolLens` contract. The returned values from this method currently assume rates are block based (`supplyRatePerBlock`, `borrowRatePerBlock`).
2) We fetch hypothetical supply and borrow rates based on pre-defined parameters in order to create a graph showing the interest rate model used (e.g.: https://app.venus.io/#/isolated-pools/pool/0x1b43ea8622e76627B81665B1eCeBB4867566B963/market/0x9f2FD23bd0A5E08C5f2b9DD6CF9C96Bfb5fA515C?chainId=56).
<img width="878" alt="Screen Shot 2024-04-04 at 11 01 42" src="https://github.com/VenusProtocol/isolated-pools/assets/44675210/8e339ac7-0e44-4457-9f63-a9579f3baf32">


In both cases, we need to have a way to know whether the rates returned are block or time based. I understand having to change all vToken contracts deployed on other chains should be avoided.

In case 1) we could perhaps update the `VTokenMetadata` struct, while in case 2) we could instead use a function from the interest rate model to know what mechanism is being used (`isTimeBased()`)?

Reply:
1) Yes that works.

2) Could we upgrade/redeploy the `PoolLens` and `IRM` contracts on other chains to add the `isTimeBased` public variable to them? I understand this is more work on your side, but my understanding is that we'll anyway need to at least update the `PoolLens` contracts since some props of `VTokenMetaData` will be renamed.

Reply:
For 2. For `PoolLens` we can redeploy it on all chains that will be doable. But for the case of `IRM` as this contract is non-upgradeable, along with it most of the `VTokens` have their own `IRM`. Because of it we will have to deploy many `IRMs` corresponding to each market on each chain. I do not think we should do it. We will have to find another way to handle it.
Wdys @chechu ?

Reply:
I think the `PoolLens` update alone is fine, since that' the primary use we have (and my main concern in terms of performances).

For the Market page (where we use `getSupplyRate` and `getBorrowRate` from the `IRM` contract), we'll just call the `isTimeBased` function from the `PoolLens`.

Reply:
Sounds good, just need an opinion from @chechu?

Reply:
The idea of adding the `isTimeBased` flag to the PoolLens contract lgtm